src/hssm/plotting/utils.py

```diff
@@ -0,0 +1,105 @@
+"""Plotting utilities for HSSM."""
+
+import logging
+
+import arviz as az
+import pandas as pd
+import xarray as xr
+
+from ..utils import _random_sample
+
+_logger = logging.getLogger("hssm")
+
+
+def _xarray_to_df(
+    posterior: xr.DataArray, n_samples: int | float | None
+) -> pd.DataFrame:
+    """Extract samples from posterior and converts it to a DataFrame.
+
+    We make the following assumptions:
+    1. The inference data always has a posterior predictive group with a `rt,response`
+       variable.
+    2. This variable always has four dimensions: `chain`, `draw`, `rt,response_obs`,
+        and `rt,response_dim`.
+
+    Parameters
+    ----------
+    idata
+        An InferenceData object.
+    n_samples
+        When an interger >= 1, the number of samples to be extracted from the draw
+        dimension. When a float between 0 and 1, the proportion of samples to be
+        extracted from the draw dimension. When None, all samples are extracted.
+
+    Returns
+    -------
+    pd.DataFrame
+        A dataframe with the posterior samples.
+    """
+    sampled_posterior = _random_sample(posterior, n_samples=n_samples)
+
+    # Convert the posterior samples to a dataframe
+    stacked = (
+        sampled_posterior.stack({"obs": ["chain", "draw", "rt,response_obs"]})
+        .transpose()
+        .to_pandas()
+        .sort_index(axis=0, level=["chain", "draw", "rt,response_obs"])
+        .droplevel(["chain", "draw"], axis=0)
+    )
+
+    # Rename the columns
+    stacked.columns = ["rt", "response"]
+
+    return stacked
+
+
+def _get_plotting_df(
```

Comment:
Note that the definition of this function intersect with a comment I make above in the context of the posterior_predictive plot. 

The addition or "real" data, should be optional.

Reply:
Cool. Will work on this